lib/Oscillator.js

```diff
@@ -16,7 +28,69 @@ var Oscillator = Layer.extend(oscillatorParams, {
     oscillator.frequency.value = frequency || 0;
 
     oscillator.detune.value = params.pitch * 100;
-    oscillator.type = params.shape;
+
+    if (params.shape != 'custom'){
+      oscillator.type = params.shape;
+    }
+    else{
+
+      var ac, wave, real, imag;
+
+      if (!window.bap)
+          window.bap = {};
+
+      if (!window.bap.ac){
```

Comment:
There is no need to create any additional audio contexts - every instance of a subclass of `bap.Model` will have the singleton audio context as `this.context`